### ISSUE TITLE
How to implement 1F1B pipeline parallelism in Jax?

### ISSUE THREAD

**Issue by MoFHeka** [google/maxtext#752]
Not GPipe. Run pipeline forward meanwhile backward.

**Comment by gobbleturk** [comment]
We are still looking into this in the open source side! Likely at least 6 months away

**Comment by sbhavani** [comment]
Is this still in progress for maxtext?

**Comment by gobbleturk** [comment]
Yes this is still something we are considering, likely requires a paradigm shift (multiple programs multiple data) to support

**Comment by gobbleturk** [comment]
One way to implement this in jax is using mpmd partir/shardy - see https://github.com/openxla/shardy

We will work on integrating this into maxtext but you can use this as a reference
